longclaw/basket/api.py

```diff
@@ -66,24 +71,33 @@ def bulk_update(self, request):
         serializer = BasketItemSerializer(self.get_queryset(request), many=True)
         response = Response(data=serializer.data,
                             status=status.HTTP_200_OK)
+        
+        basket_modified.send(sender=BasketItem, basket_id=bid)
+        
         return response
 
     def destroy(self, request, variant_id=None):
         """
         Remove an item from the basket
         """
+        bid = utils.basket_id(request)
+        
         variant = ProductVariant.objects.get(id=variant_id)
         quantity = int(request.data.get("quantity", 1))
         try:
             item = BasketItem.objects.get(
-                basket_id=utils.basket_id(request), variant=variant)
+                basket_id=bid, variant=variant)
             item.decrease_quantity(quantity)
         except BasketItem.DoesNotExist:
             pass
 
         serializer = BasketItemSerializer(self.get_queryset(request), many=True)
-        return Response(data=serializer.data,
+        response = Response(data=serializer.data,
                         status=status.HTTP_200_OK)
+        
+        basket_modified.send(sender=BasketItem, basket_id=bid)

```

Comment:
Should the modified signal be sent here? The try/catch block lets BasketItems which do not exist silently pass - so we can send an OK response without modifying the basket at all. Do we still need the signal to say the basket was modified in this case?

Reply:
If we don't then we need a basket destroyed signal so the [clear_basket_rates()](https://github.com/JamesRamm/longclaw/pull/230/files#diff-3fcda2aa2b0f0ac1a559edf34571c82eR132) receiver can remove stale rate records